tests/integration/test_bayesian_optimization.py

```diff
@@ -349,7 +349,7 @@ def test_bayesian_optimizer_with_sgpr_finds_minima_of_simple_quadratic() -> None
             25,
             EfficientGlobalOptimization(
                 ParallelContinuousThompsonSampling(),
-                num_query_points=3,
```

Comment:
How does this change (and the change in the parameters of `_test_optimizer_finds_minimum` above) relate to only running the slow tests?

Reply:
These are tests that were previously not being run due to the OOM crash. Removing the crash showed that the tests were fragile: each was passing in one environment but failing in another. The changes are aiming to make them a bit less fragile and pass in both (which they now do: https://github.com/secondmind-labs/trieste/actions/runs/4045012501).

Reply:
(PS You can see that both the changed tests were decorated with `@pytest.mark.slow`)